### ISSUE TITLE
Working with firebase in flutter Add to App feature

### ISSUE THREAD

**Issue by Kabilan21** [FirebaseExtended/flutterfire#6667]
### Discussed in https://github.com/FirebaseExtended/flutterfire/discussions/6666

<div type='discussions-op-text'>

<sup>Originally posted by **Kabilan21** July 21, 2021</sup>
hi , I am trying to add my flutter app to an android app,I use firebase from both main activity and inside flutter app.How to initialize firebase in this case. can anyone help?</div>

E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)
    #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156:7)
    <asynchronous suspension>
    #1      MethodChannel.invokeListMethod (package:flutter/src/services/platform_channel.dart:344:35)
    <asynchronous suspension>
    #2      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:30:23)
    <asynchronous suspension>
    #3      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:77:7)
    <asynchronous suspension>
    #4      Firebase.initializeApp (package:firebase_core/src/firebase.dart:41:31)
    <asynchronous suspension>

i am receiving the above error from flutter module while trying to Firebase.initializeApp();

**Comment by markusaksli-nc** [comment]
Hi @Kabilan21 
Could you please provide your `flutter doctor -v` and `flutter pub deps -- --style=compact` of your flutter module?
Thank you

**Comment by Kabilan21** [comment]
@markusaksli-nc Thank you for your quick response.



**flutter doctor -v** :

[√] Flutter (Channel stable, 2.0.2, on Microsoft Windows [Version 10.0.18363.1379], locale en-US)
    • Flutter version 2.0.2 at C:\Users\kabil\Documents\flutter
    • Framework revision 8962f6dc68 (4 months ago), 2021-03-11 13:22:20 -0800
    • Engine revision 5d8bf811b3
    • Dart version 2.12.1

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\kabil\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.58.2)
    • VS Code at C:\Users\kabil\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.24.0

[√] Connected device (1 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86 • Android 11 (API 30) (emulator)

• No issues found!


**flutter pub deps -- --style=compact :**
Dart SDK 2.12.1
Flutter SDK 2.0.2
learn_module 1.0.0+1

dependencies:
- cupertino_icons 1.0.3
- firebase_core 1.4.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 10.0.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_remote_config 0.10.0+3 [firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- path_provider 2.0.2 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.5.0 [collection]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.1.0
- charcode 1.2.0
- clock 1.1.0
- collection 1.15.0
- fake_async 1.2.0 [clock collection]
- ffi 1.1.2
- file 6.1.2 [meta path]
- firebase_core_platform_interface 4.0.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.1.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 3.0.3 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.0.3 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_remote_config_platform_interface 0.3.0+3 [firebase_core flutter meta plugin_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- js 0.6.3
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- path 1.8.0
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 2.0.0 [flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.1 [path_provider_platform_interface meta path flutter ffi win32]
- platform 3.0.0
- plugin_platform_interface 2.0.1 [meta]
- process 4.2.1 [file path platform]
- sky_engine 0.0.99
- source_span 1.8.0 [charcode collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- vector_math 2.1.0
- win32 2.0.5 [ffi]
- xdg_directories 0.2.0 [meta path process]

Below are the error while run the flutter module as a fragment in android app:

E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)
    #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156:7)
    <asynchronous suspension>
    #1      MethodChannel.invokeListMethod (package:flutter/src/services/platform_channel.dart:344:35)
    <asynchronous suspension>
    #2      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:30:23)
    <asynchronous suspension>
    #3      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:77:7)
    <asynchronous suspension>
    #4      Firebase.initializeApp (package:firebase_core/src/firebase.dart:41:31)
    <asynchronous suspension>
    #5      main (package:learn_module/main.dart:7:3)
    <asynchronous suspension>
D/FRCPlugin: Sending fetchTimeout: 60
E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)
    #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156:7)
    <asynchronous suspension>
    #1      MethodChannel.invokeListMethod (package:flutter/src/services/platform_channel.dart:344:35)
    <asynchronous suspension>
    #2      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:30:23)
    <asynchronous suspension>
    #3      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:77:7)
    <asynchronous suspension>
    #4      Firebase.initializeApp (package:firebase_core/src/firebase.dart:41:31)
    <asynchronous suspension>
    #5      main (package:learn_module/main.dart:7:3)
    <asynchronous suspension>

**Comment by markusaksli-nc** [comment]
First off I would recommend you `flutter upgrade` to the latest version of stable. It won't solve the issue but it will get you up to date with all of the latest fixes and improvements in general.

The reason you are getting this issue is because the method channel isn't being properly established between the on-device native Firebase SDK and Flutter. The FlutterFire plugins simply wrap the native Firebase SDKs so they could be called from Dart via method channels. These dependencies are usually internally managed by the plugins but this may not work in an a2a scenario.

What setup steps have you followed to use the FlutterFire plugins so far?

As a reminder:
https://firebase.flutter.dev/docs/installation/android

**Comment by Kabilan21** [comment]
I found that the problem is not related to firebase and the error occurs for various plugins. So I will create an issue in the flutter channel. As you said, can't we use a2a with a flutter module having dependencies containing platform method channels.

can you try creating a simple a2a with android and flutter module with a plugin(ex: uni_links: ^0.5.1 (uses platform method channels ))?

**Comment by Kabilan21** [comment]
I followed everything described in the add-to-app documentation provided in the flutter website.I manually created flutter module and linked to the android app .I was able to work with the firebase inside the flutter module successfully.Previously i linked the flutter module to android app using android studio instead of manual linking by typing in commands.

one can try the  [sample app](https://github.com/flutter/samples/tree/master/add_to_app/plugin) provided by the flutter team for reference.The problem is solved.is it a good choice to write some part of flutter app in native for performance reasons. 
Like i can write a single activity to implement recycler view instead of listview builder in flutter which has scrolling issues (in case of list view of images).
